src/main/java/app/accommodationbookingservice/service/impl/PaymentServiceImpl.java

```
@@ -0,0 +1,127 @@
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional
+public class PaymentServiceImpl implements PaymentService {
+    private final PaymentRepository paymentRepo;
+    private final BookingRepository bookingRepo;
+    private final AccommodationRepository accommodationRepo;
+    private final String stripeKey;
+
+    public PaymentServiceImpl(PaymentRepository paymentRepo,
+                              BookingRepository bookingRepo,
+                              AccommodationRepository accommodationRepo,
+                              @org.springframework.beans.factory
+                                      .annotation.Value("${stripe.secret-key}") String stripeKey) {
+        this.paymentRepo = paymentRepo;
+        this.bookingRepo = bookingRepo;
+        this.accommodationRepo = accommodationRepo;
+        this.stripeKey = stripeKey;
+    }
+
+    @Override
+    public Payment createSession(Long bookingId, String successUrl, String cancelUrl) {
+        Stripe.apiKey = stripeKey;
+
+        Booking booking = bookingRepo.findById(bookingId)
+                .orElseThrow(() -> new EntityNotFoundException("Booking not found: "
+                        + bookingId));
+        Accommodation acc = accommodationRepo.findById(booking.getAccommodationId())
+                .orElseThrow(() -> new EntityNotFoundException("Accommodation not found: "
+                        + booking.getAccommodationId()));
+        long days = ChronoUnit.DAYS.between(booking.getCheckInDate(), booking
+                .getCheckOutDate());
+        BigDecimal amount = acc.getDailyRate().multiply(BigDecimal.valueOf(days));
+
+        SessionCreateParams.LineItem.PriceData priceData =
+                SessionCreateParams.LineItem.PriceData.builder()
+                        .setCurrency("usd")
+                        .setUnitAmount(amount.multiply(BigDecimal.valueOf(100)).longValue())
```

Converting `BigDecimal `to cents via `.longValue()` can silently truncate fractions. Use `.longValueExact()` or apply explicit rounding to avoid data loss.